back/engines/free/email_campaigns/app/models/email_campaigns/campaigns/user_digest.rb

```diff
@@ -96,9 +94,9 @@ def generate_commands(recipient:, time: nil)
 
     # @return [Boolean]
     def content_worth_sending?(_)
-      @content_worth_sending ||= TrendingIdeaService.new.filter_trending(
-        IdeaPolicy::Scope.new(nil, Idea).resolve.where(publication_status: 'published')
-      ).count('*') >= N_TOP_IDEAS
+      # Check positive? as fetching a non-integer env var would result in zero and this hook would return true,
+      # whilst top_ideas would be limited to zero ideas, possibly resulting in no content being sent.
```

Comment:
This wasn't causing the bug, but it could.

Reply:
Very good remark. Probably, we could even go further and make all these constants static like `N_TOP_IDEAS = 3`. If we didn't set it dynamically (via an ENV var) for 5 years, chances are we'll never be setting it.

Reply:
Yes. I learned a lot about mails in general from this bug hunt.

We should also review all mail models for any filtering done in the payload that might conflict with the before hook / filtering done before the payload.

Reply:
Will you add a ticket to make the constants static? Or will you do it in this PR? Or do you think it's not worth it?

Reply:
I created a ticket for this, som we don't forget: https://citizenlab.atlassian.net/browse/CL-3348